service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AdditionalApplicationsUploadedEmailContentProvider.java

```diff
@@ -0,0 +1,121 @@
+
+        return applicationTypes;
+    }
+
+    private void getC2DocumentBundleApplicationTypes(List<String> applicationTypes, C2DocumentBundle c2DocumentBundle) {
+        if (c2DocumentBundle == null) {
+            return;
+        }
+
+        String c2Type = String.format("C2 (%s)", c2DocumentBundle.getType().getLabel());
+        List<C2AdditionalOrdersRequested> c2AdditionalOrdersRequested =
+            c2DocumentBundle.getC2AdditionalOrdersRequested();
+
+
+
+        if (isNotEmpty(c2AdditionalOrdersRequested)) {
+            List<String> c2AdditionalOrdersRequestedLabels = new ArrayList<>();
+
+            for (C2AdditionalOrdersRequested orderRequested : c2AdditionalOrdersRequested) {
+                if (C2AdditionalOrdersRequested.PARENTAL_RESPONSIBILITY.equals(orderRequested)) {
+                    c2AdditionalOrdersRequestedLabels.add(c2DocumentBundle.getParentalResponsibilityType().getLabel());
+                } else {
+                    c2AdditionalOrdersRequestedLabels.add(orderRequested.getLabel());
+                }
+            }
+
+            c2Type += " - " + String.join(", ", c2AdditionalOrdersRequestedLabels);
+        }
+
+        applicationTypes.add(c2Type);
+
+        c2DocumentBundle.getSupplementsBundle().stream()
+            .map(Element::getValue)
+            .forEach(Supplement -> {
+                String supplementName = Supplement.getName().getLabel();
+                if (SupplementType.C20_SECURE_ACCOMMODATION.equals(Supplement.getName())) {
+                    supplementName += String.format(" (%s)", Supplement.getSecureAccommodationType().getLabel());
+                }
+                applicationTypes.add(supplementName);
+            });
+    }
+
+    private void getOtherApplicationsBundleApplicationTypes(List<String> applicationTypes,
+                                                            OtherApplicationsBundle otherDocumentBundle) {
+        if (otherDocumentBundle == null) {
+            return;
+        }
+
+        OtherApplicationType otherApplicationType = otherDocumentBundle.getApplicationType();
+
+        if (OtherApplicationType.C1_PARENTAL_RESPONSIBILITY.equals(otherApplicationType)) {
+            applicationTypes.add(String.format("C1 - %s",
+                otherDocumentBundle.getParentalResponsibilityType().getLabel()));
+        } else {
+            applicationTypes.add(otherDocumentBundle.getApplicationType().getLabel());
+        }
+
+        otherDocumentBundle.getSupplementsBundle().stream()
+            .map(Element::getValue)
+            .forEach(Supplement -> {
+                String supplementName = Supplement.getName().getLabel();
+                if (SupplementType.C20_SECURE_ACCOMMODATION.equals(Supplement.getName())) {
+                    supplementName += String.format(" (%s)", Supplement.getSecureAccommodationType().getLabel());
+                }
+                applicationTypes.add(supplementName);
+            });
+    }
```

Comment:
similar code used here as in line 79, thoughts on moving it to a private common method? like : 

`    private void addSupplementTypes(List<String> applicationTypes, List<Element<Supplement>> supplements) {
        supplements.stream()
            .map(Element::getValue)
            .forEach(Supplement -> {
                String supplementName = Supplement.getName().getLabel();
                if (SupplementType.C20_SECURE_ACCOMMODATION.equals(Supplement.getName())) {
                    supplementName += String.format(" (%s)", Supplement.getSecureAccommodationType().getLabel());
                }
                applicationTypes.add(supplementName);
            });
    }`

and then from each of these methods we can call like         
addSupplementTypes(applicationTypes, otherDocumentBundle.getSupplementsBundle());